code/game/objects/items/reagent_containers/glass.dm

```diff
@@ -227,6 +229,80 @@
 		var/image/lid = image(icon, src, "lid_[initial(icon_state)]")
 		overlays += lid
 
+/obj/item/reagent_container/glass/minitank
+	name = "MS-11 Smart Refill Tank"
+	desc = "A robust little tank capable of refilling autoinjectors that previously required a nanomed system to refill. Using the wonders of microchips, it automatically sorts the correct chemicals into most single reagent autoinjectors. It is unable to partially fill them however. A valve exists on the top to transfer reagents to another container or to flush it entirely."
+	icon = 'icons/obj/items/tank.dmi'
+	icon_state = "mini_reagent_tank"
+	matter = list("metal" = 500)
+	attack_speed = 4
+	volume = 180
+	w_class = SIZE_MEDIUM
+	splashable = FALSE
+	can_be_placed_into = list(
+		/obj/structure/machinery/chem_master/,
+		/obj/structure/machinery/chem_dispenser/,
+		/obj/structure/machinery/reagentgrinder,
+		/obj/structure/surface/table,
+		/obj/structure/closet,
+		/obj/structure/sink,
+		/obj/item/storage,
+		/obj/item/clothing,
+		/obj/structure/machinery/bot/medbot,
+		/obj/structure/machinery/computer/pandemic,
+		/obj/item/storage/secure/safe,
+		/obj/structure/machinery/iv_drip,
+		/obj/structure/machinery/disposal,
+		/obj/structure/machinery/sleeper,
+		/obj/structure/machinery/smartfridge/,
+		/obj/structure/machinery/biogenerator,
+		/obj/structure/machinery/reagent_analyzer,
+		/obj/structure/machinery/centrifuge,
+		/obj/structure/machinery/autodispenser,
+		/obj/structure/machinery/constructable_frame)
+
+/obj/item/reagent_container/glass/minitank/on_reagent_change()
+	update_icon()
+
+
+/obj/item/reagent_container/glass/minitank/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/reagent_container/hypospray/autoinjector))
+		var/obj/item/reagent_container/hypospray/autoinjector/A = W
+		if(A.mixed_chem)
+			to_chat(user, SPAN_WARNING("The autoinjector doesn't fit into the [src]'s valve. It's probably not compatible."))
+			return
+		if(reagents.has_reagent(A.chemname, A.volume))
+			reagents.trans_id_to(A, A.chemname, A.volume)
+			A.uses_left = 3
+			A.update_icon()
+			playsound(src.loc, 'sound/effects/refill.ogg', 25, 1, 3)
+		else
+			to_chat(user, SPAN_WARNING("A small LED on \the [src] blinks. The tank is either empty or not compatible with that!"))
+			. = ..()
+			return
+		to_chat(user,SPAN_INFO("You successfully refill \the [W.name] with \the [src]!"))
+
+/obj/item/reagent_container/glass/minitank/verb/flush_tank(mob/user)
+	if(!usr || usr.stat || usr.lying)	return
+	set category = "Object"
+	set name = "flush tank"
+	set src in view(1)
+	playsound(src.loc, 'sound/effects/refill.ogg', 25, 1, 3)
+	to_chat(user, SPAN_WARNING("You work the flush valve and successfully flush \the [src]'s contents!"))
+	reagents.clear_reagents()
+	return
```

Comment:
![image](https://user-images.githubusercontent.com/66756236/166210214-0c1b40e5-39e5-44f2-bab0-afb16ce3ed1d.png)

yeah you definitely need to fix this verb lol